src/battle_ai_switch_items.c

```diff
@@ -228,44 +217,177 @@ static bool8 FindMonThatAbsorbsOpponentsMove(void)
     return FALSE;
 }
 
-static bool8 ShouldSwitchIfNaturalCure(void)
+static bool8 ShouldSwitchIfGameStatePrompt(void)
 {
-    if (!(gBattleMons[gActiveBattler].status1 & STATUS1_SLEEP))
-        return FALSE;
-    if (AI_GetAbility(gActiveBattler) != ABILITY_NATURAL_CURE)
-        return FALSE;
-    if (gBattleMons[gActiveBattler].hp < gBattleMons[gActiveBattler].maxHP / 2)
-        return FALSE;
+    bool8 switchMon = FALSE;
+    u16 monAbility = AI_GetAbility(gActiveBattler);
```

Comment:
we compute all AI abilities, hold effects, etc before the turn selection sequence, so you can do `u16 monAbility = AI_DATA->abilities[gActiveBattler]` and `u16 holdEffect = AI_DATA->holdEffects[gActiveBattler]` instead.